src/server.jl

```diff
@@ -436,25 +443,23 @@ function serve(fw::FileWatcher=SimpleWatcher(file_changed_callback);
     start(fw)
 
     # make request handler
-    req_handler = HTTP.RequestHandlerFunction() do req
+    req_handler = streamhandler() do req
         req = preprocess_request(req)
         serve_file(fw, req; inject_browser_reload_script = inject_browser_reload_script, allow_cors = allow_cors)
     end
 
     server = Sockets.listen(parse(IPAddr, host), port)
     url = "http://$(host == string(Sockets.localhost) ? "localhost" : host):$port"
     println("✓ LiveServer listening on $url/ ...\n  (use CTRL+C to shut down)")
-    @async HTTP.listen(host, port;
-                       server=server, readtimeout=0, reuse_limit=0) do http::HTTP.Stream
-        # reuse_limit=0 ensures that there won't be an error if killing and restarting the server.
-        if HTTP.WebSockets.is_upgrade(http.message)
+    HTTP.listen!(host, port; server=server, readtimeout=0) do http::HTTP.Stream
```

Comment:
Note we should be able to use the simpler
```suggestion
    HTTP.listen!(server; readtimeout=0) do http::HTTP.Stream
```
form now which avoids repeating the host/port args so much.

Reply:
Great, that has annoyed me quite a bit in the past.